src/api/wallet_rpc.browserextension.ts

```diff
@@ -0,0 +1,87 @@
+import { AWCStreamChannel, AWCStreamChannelState, ArchethicWalletClient } from "./wallet_rpc.js"
+
+
+declare global {
+  /**
+   * Objects injected by ArchethicWallet browser extension.
+   */
+  const archethic: {
+    streamChannel: AWCStreamChannel<string> | undefined,
+    awc: ArchethicWalletClient | undefined,
+  } | undefined
+}
+
+export class AWCWebBrowserExtension {
+  static get awc(): ArchethicWalletClient | undefined {
+    return (typeof (archethic) === "undefined") ? undefined : archethic?.awc
+  }
+}
+
+export class AWCWebBrowserExtensionStreamChannel implements AWCStreamChannel<string> {
+  private extensionId: string
+  private _port: chrome.runtime.Port | null = null
+  private _state: AWCStreamChannelState = AWCStreamChannelState.CLOSED
+
+  constructor(extensionId: string | undefined) {
+    if (extensionId === undefined) throw new Error('Archethic Wallet Web extension not available')
+    this.extensionId = extensionId
+  }
+
+  async connect(): Promise<void> {
+    if (this._port !== null) {
+      console.log(`[AWC] Popup extension already running`)
+      return
+    }
+
+    this._state = AWCStreamChannelState.CONNECTING
+
+    console.log(`[AWC] Wait for popup extension ...`)
+    await chrome.runtime.sendMessage(this.extensionId, 'ensureExtensionPopupOpened')
```

Comment:
I saw on MDN doc it's prefered to support cross platform api like suggest by:
> Firefox also supports the chrome.* namespace for APIs that are compatible with Chrome, primarily to assist with [porting](https://extensionworkshop.com/documentation/develop/porting-a-google-chrome-extension/). However, using the browser.* namespace is preferred. In addition to being the proposed standard, browser.* uses promises—a modern and convenient mechanism for handling asynchronous events.

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Build_a_cross_browser_extension

Maybe using a polyfill or using the browser.* can leverage firefox & chrome at the same time

Reply:
`browser.*` is not supported by chrome yet. 
Using a polyfill like [this one ](https://github.com/mozilla/webextension-polyfill) seems like a requirement to get a wide compatibility. 

I'll look at that in a second step 👍